src/fmeval/model_runners/composers/jumpstart_composer.py

```diff
@@ -12,17 +14,23 @@ class JumpStartComposer(Composer):
     Jumpstart model request composer
     """
 
-    def __init__(self, jumpstart_model_id: str, jumpstart_model_version: str):
+    def __init__(
+        self, jumpstart_model_id: str, jumpstart_model_version: str, is_embedding_model: Optional[bool] = False
+    ):
         """
         Initialize the JumpStartComposer for the given JumpStart model_id and model_version.
         """
         self._model_id = jumpstart_model_id
         self._model_version = jumpstart_model_version
+        self._is_embedding_model = is_embedding_model
 
     def compose(self, prompt: str) -> JumpStartSerializablePayload:
         """
         Composes the payload for the given JumpStartModel from the provided prompt.
         """
+        # embedding models take raw text as input payload
+        if self._is_embedding_model:
+            return prompt
```

Comment:
Based on the signature of this method, you should still return an object of type `JumpStartSerializablePayload`. Can you construct such a payload with just the `prompt` data? I would rather stay away from changing the signature to `Union[JumpStartSerializablePayload, str]` and directly returning `prompt` if possible.

Reply:
Per offline discyssion change to `Union[JumpStartSerializablePayload, bytes]` as won't be able to construct payload.